group-coordinator/src/test/java/org/apache/kafka/coordinator/group/GroupMetadataManagerTest.java

```diff
@@ -360,6 +368,27 @@ public void testStreamsRequestValidation() {
                 .setRackId("")));
         assertEquals("RackId can't be empty.", ex.getMessage());
 
+        // InstanceId can't be null with static membership
+        ex = assertThrows(InvalidRequestException.class, () -> context.streamsGroupHeartbeat(
+                new StreamsGroupHeartbeatRequestData()
+                        .setGroupId("foo")
+                        .setMemberId(memberId)
+                        .setMemberEpoch(LEAVE_GROUP_STATIC_MEMBER_EPOCH)
+                        .setInstanceId(null)
+                        .setRackId("rackid")));
+        assertEquals("InstanceId can't be null.", ex.getMessage());
+
+        // With non-static membership, the memberEpoch must be -1 or greater
```

Comment:
This comment is a bit misleading because this verification verifies that the member epoch is greater or equal -2 no matter whether static membership is used or not.